File: swift/Ice/greeter/Sources/GreeterServer/Chatbot.swift
// Copyright (c) ZeroC, Inc.

import Ice

class ChatBot: Greeter {
    func greet(name: String, current _: Ice.Current) throws -> String {
        print("Dispatching greet request { name = '\(name)' }")
        return "Hello, \(name)!"
    }
}

File: python/Ice/greeter/server/chatbot.py
# Copyright (c) ZeroC, Inc.

import Ice

# Slice module VisitorCenter in Greeter.ice maps to Python module VisitorCenter.
import VisitorCenter

class Chatbot(VisitorCenter.Greeter):
    def greet(self, name: str, current: Ice.Current) -> str:
        print(f"Dispatching greet request {{ name = '{name}' }}")
        return f"Hello, {name}!"
